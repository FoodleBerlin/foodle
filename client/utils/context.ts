import jwt from 'jsonwebtoken';

export const extractUserFromToken = (rawCookieToken: string | null, rawHeaderToken: string): any | null => {
  if (!rawHeaderToken && !rawCookieToken) return null;
  const decoded = jwt.verify(
    rawCookieToken ?? rawHeaderToken,
    process.env.SERVER_SECRET ?? '',
    (err: any, data: any) => {
      if (!err && data.user.id) {
        return data.user;
      } else {
        return null;
      }
    }
  );
  // JWT verify lib functionr returns void
  return decoded as unknown as any;
};
