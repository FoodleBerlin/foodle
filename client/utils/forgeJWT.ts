
export type Token = {
  user: {
    id: string;
    fullName: string;
    email: string;
    stripeId: string | null;
  };
};

