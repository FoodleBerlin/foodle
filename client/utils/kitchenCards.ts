import { KitchenCardInfo } from '../../client/components/Landing/SwiperCard';

export const kitchenCards: KitchenCardInfo[] = [
  {
    title: 'Industrial Grade Kitchen Mitte',
    description: 'Practically New Kitchen in the center of the center. Come Join our...',
    tags: ['Verified', 'Oven', 'Skillet', 'Freezer', 'Fridge', '...'],
    cityRegion: 'Mitte',
    distance: 5.1,
    price: 50,
  },
  {
    title: 'Cozy 3 person kitchen Moabit',
    description: 'Small kitchen with not so much equipment, but an opportunity for...',
    tags: ['Verified', 'Oven', 'Fridge', 'Hot Plate', 'Air Fryer', '...'],
    cityRegion: 'Moabit',
    distance: 3.2,
    price: 40,
  },
  {
    title: 'Pizzeria Kitchen Friedrichshain',
    description: 'Fully stocked kitchen in a very popular area. Please take a loot at...',
    tags: ['Shared', 'Verified', 'Pizza Oven', 'Skillet', '...'],
    cityRegion: 'Friedrichshain',
    distance: 4.3,
    price: 19,
  },
  {
    title: 'Sushi Kitchen in Reinickendorf',
    description: 'Sushi kitchen open for 2 days a week, very well equipped and flexible... ',
    tags: ['Verified', 'Oven', 'Skillet', 'Freezer', 'Fridge', '...'],
    cityRegion: 'Reinickendorf',
    distance: 6.5,
    price: 32,
  },
  {
    title: 'Bakery Open 7 evenings a Week ',
    description: 'Our bakery kitchen is usable from 1pm onwards, please be careful with...',
    tags: ['Verified', 'Dual-Oven', 'Oven', 'Fridge', '...'],
    cityRegion: 'Kreuzberg',
    distance: 1.1,
    price: 60,
  },
  {
    title: 'Thai kitchen for rent Neukölln',
    description: 'Newly opened Thai kitchen available on Mondays and Tuesdays...',
    tags: ['Shared', 'Verified', 'Oven', 'Skillet', 'Deep-Fryer', '...'],
    cityRegion: 'Neukölln',
    distance: 5.1,
    price: 25,
  },
];
