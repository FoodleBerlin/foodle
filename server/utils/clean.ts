import prisma from '../singletons/prisma';

export async function clean() {
  try {
    await prisma.$executeRaw`DELETE FROM "User";`;
  } catch (e) {
    throw new Error(`Unable to seed ${e}`);
  }
}
