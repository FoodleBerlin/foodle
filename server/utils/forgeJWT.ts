import { User } from '@prisma/client';
import jwt from 'jsonwebtoken';
import prisma from '../singletons/prisma';

export type Token = {
  user: {
    id: string;
    fullName: string;
    email: string;
    stripeId: string | null;
  };
};
export default async function forgeJWT(user: Pick<User, 'id' | 'fullName' | 'email' | 'stripeId'>) {
  const id = await prisma.user.findUnique({
    where: { id: user.id },
  });
  if (!id) throw new Error('No id found for test user');
  const tokenRaw: Token = {
    user: {
      id: user.id,
      fullName: user.fullName,
      email: user.email,
      stripeId: user.stripeId,
    },
  };
  const token = jwt.sign(tokenRaw, process.env.SERVER_SECRET ?? '', {
    expiresIn: '1d',
  });
  return token;
}
