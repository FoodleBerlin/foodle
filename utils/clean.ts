import prisma from "../server/singletons/prisma";
export async function clean() {
  try {
    await prisma.$executeRaw`DELETE FROM "Booking";`;
    await prisma.$executeRaw`DELETE FROM "Property";`;
    await prisma.$executeRaw`DELETE FROM "User";`;
  } catch (e) {
    throw new Error(`Unable to seed ${e}`);
  }
}
