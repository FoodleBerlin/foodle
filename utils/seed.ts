import prisma from '../server/singletons/prisma';
export async function seed() {
  const [users] = await Promise.all(
    await prisma.$transaction([
      prisma.user.createMany({
        data: [
          {
            id: '1',
            email: 'user1@gmail.com',
            handle: 'user1',
            fullName: 'User 1',
            role: 'tenant',
            zip: 13000,
          },
          {
            id: '2',
            email: 'use2@gmail.com',
            handle: 'user2',
            fullName: 'User 2',
            role: 'tenant',
            zip: 14000,
          },
          {
            id: '3',
            email: 'user3@gmail.com',
            handle: 'user3',
            fullName: 'User 3',
            role: 'tenant',
            zip: 14000,
          },
          {
            id: '4',
            email: 'user4@gmail.com',
            handle: 'user4',
            fullName: 'User 4',
            role: 'landlord',
            zip: 15000,
          },
          {
            id: '5',
            email: 'user5@gmail.com',
            handle: 'user5',
            fullName: 'User 5',
            role: 'tenantLandlord',
            zip: 16000,
          },
        ],
      }),
    ])
  );
  await prisma.property.createMany({
    data: [
      {
        id: '1',
        size: 123,
        ownerId: '1',
        street: 'testhbdbjkfsdbjf',
        streetNumber: 1233,
        zip: 123445,
        city: 'berlin',
        description: 'testhbdbjkfsdbjf',
        rules: 'fffff',
        dailyPrice: 100,
        serviceFee: 0,
      },
    ],
  });
}
export default seed;
