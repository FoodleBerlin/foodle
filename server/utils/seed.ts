import prisma from '../singletons/prisma';
import { clean } from './clean';
export async function seed() {
  await clean();
  const [users] = await Promise.all(
    await prisma.$transaction([
      prisma.user.createMany({
        data: [
          {
            id: '1',
            email: 'user1@gmail.com',
            handle: 'user1',
            fullName: 'User 1',
            role: 'tenant',
            zip: 13000,
            description: 'desc',
            passportS3Id: 'passS3',
            solvencyS3Id: 'solvS3',
            licenseS3Id: 'liceS3',
          },
          {
            id: '2',
            email: 'use2@gmail.com',
            handle: 'user2',
            fullName: 'User 2',
            role: 'tenant',
            zip: 14000,
            description: 'desc',
            passportS3Id: 'passS3',
            solvencyS3Id: 'solvS3',
            licenseS3Id: 'liceS3',
          },
          {
            id: '3',
            email: 'user3@gmail.com',
            handle: 'user3',
            fullName: 'User 3',
            role: 'tenant',
            zip: 14000,
            description: 'desc',
            passportS3Id: 'passS3',
            solvencyS3Id: 'solvS3',
            licenseS3Id: 'liceS3',
          },
          {
            id: '4',
            email: 'user4@gmail.com',
            handle: 'user4',
            fullName: 'User 4',
            role: 'landlord',
            zip: 15000,
            description: 'desc',
            passportS3Id: 'passS3',
            solvencyS3Id: 'solvS3',
            licenseS3Id: 'liceS3',
          },
          {
            id: '5',
            email: 'user5@gmail.com',
            handle: 'user5',
            fullName: 'User 5',
            role: 'tenantLandlord',
            zip: 16000,
            description: 'desc',
            passportS3Id: 'passS3',
            solvencyS3Id: 'solvS3',
            licenseS3Id: 'liceS3',
          },
        ],
      }),
    ])
  );

  await prisma.$transaction([
    prisma.property.createMany({
      data: [
        {
          zip: 1355,
          size: 123,
          street: 'sample-street',
          streetNumber: '23',
          city: 'Berlin',
          description: 'user1-listing',
          serviceFee: 5,
          rules: ['rule-1', 'rule-2'],
          title: 'prop title',
          hourlyPrice: 12,
          deposit: 2,
          images: ['test-image-url'],
          partialSpace: false,
          ownerId: '1',
          handle: 'prop1',
          frequency: 'weekly',
          facilities: ['Unfurnished'],
          startDate: '2022-06-27T08:00:00.003Z',
          endDate: '2022-06-30T08:00:00.003Z',
        },
        {
          zip: 1355,
          size: 123,
          street: 'sample-street',
          streetNumber: '23',
          city: 'Berlin',
          description: 'user1-listing',
          serviceFee: 5,
          rules: ['rule-1', 'rule-2'],
          title: 'prop title',
          hourlyPrice: 12,
          deposit: 2,
          images: ['test-image-url'],
          partialSpace: false,
          ownerId: '2',
          handle: 'prop2',
          frequency: 'weekly',
          facilities: ['Elevator', 'Parking', 'Dishwasher'],
          startDate: '2022-08-27T08:00:00.003Z',
          endDate: '2022-09-30T08:00:00.003Z',
        },
        {
          zip: 1355,
          size: 123,
          street: 'sample-street',
          streetNumber: '23',
          city: 'Berlin',
          description: 'user1-listing',
          serviceFee: 5,
          rules: ['rule-1', 'rule-2'],
          title: 'prop title',
          hourlyPrice: 12,
          deposit: 2,
          images: ['test-image-url'],
          partialSpace: false,
          ownerId: '2',
          handle: 'prop3',
          frequency: 'weekly',
          facilities: ['Elevator', 'Parking'],
          startDate: '2022-10-27T08:00:00.003Z',
          endDate: '2022-11-30T08:00:00.003Z',
        },
      ],
    }),
  ]);

  const prop1 = await prisma.property.findUnique({
    where: {
      handle: 'prop1',
    },
  });
  if (prop1 == null) {
    throw Error('Error while seeding db, property with handle prop1 does not exist.');
  }
  const prop2 = await prisma.property.findUnique({
    where: {
      handle: 'prop2',
    },
  });
  if (prop2 == null) {
    throw Error('Error while seeding db, property with handle prop2 does not exist.');
  }
  // seeding for booking request
  /*  
      test weekly, one weekday
      {
        daySlots: {
          startTime: '2022-06-27T08:00:00.003Z',
          endTime: '2022-06-27T16:00:00.003Z',
        },
        frequency: 'NONE',
        startDate: '2022-06-27T08:00:00.003Z',
        endDate: '2022-07-25T08:00:00.003Z',
        propertyHandle: 'prop1',
      } */
  await prisma.$transaction([
    prisma.daySlot.createMany({
      data: [
        {
          startTime: '2022-06-27T08:00:00.003Z',
          endTime: '2022-06-27T16:00:00.003Z',
          propertyId: prop1.id,
        },
        {
          startTime: '2022-07-04T08:00:00.003Z',
          endTime: '2022-07-04T16:00:00.003Z',
          propertyId: prop1.id,
        },
        {
          startTime: '2022-07-11T08:00:00.003Z',
          endTime: '2022-07-11T16:00:00.003Z',
          propertyId: prop1.id,
        },
        {
          startTime: '2022-07-18T08:00:00.003Z',
          endTime: '2022-07-18T16:00:00.003Z',
          propertyId: prop1.id,
        },
        {
          startTime: '2022-07-25T08:00:00.003Z',
          endTime: '2022-07-25T16:00:00.003Z',
          propertyId: prop1.id,
        },

        /* 
          test monthly + multiple weekdays + varying day time
          
          const bookingVars = {
            daySlots: [
              {
                startTime: '2022-05-23T10:00:00.003Z',
                endTime: '2022-05-23T16:00:00.003Z',
              },
              {
                startTime: '2022-05-25T08:00:00.003Z',
                endTime: '2022-05-25T16:00:00.003Z',
              },
            ],
            frequency: 'MONTHLY',
            startDate: '2022-05-25T08:00:00.003Z',
            endDate: '2022-07-22T08:00:00.003Z',
            propertyHandle: 'prop2',
          };
        */

        {
          startTime: '2022-05-25T08:00:00.003Z',
          endTime: '2022-05-25T16:00:00.003Z',
          propertyId: prop2.id,
        },
        {
          startTime: '2022-05-23T08:00:00.003Z',
          endTime: '2022-05-23T16:00:00.003Z',
          propertyId: prop2.id,
        },
        {
          startTime: '2022-06-20T08:30:00.003Z',
          endTime: '2022-06-20T16:00:00.003Z',
          propertyId: prop2.id,
        },
        {
          startTime: '2022-06-22T08:00:00.003Z',
          endTime: '2022-06-22T16:00:00.003Z',
          propertyId: prop2.id,
        },
        {
          startTime: '2022-07-18T09:00:00.003Z',
          endTime: '2022-07-18T16:00:00.003Z',
          propertyId: prop2.id,
        },
        {
          startTime: '2022-07-20T08:00:00.003Z',
          endTime: '2022-07-20T16:00:00.003Z',
          propertyId: prop2.id,
        },
        /* 
          test frequency none
          {
          "daySlots": {
            "startTime": "2022-06-27T09:00:00.003Z",
            "endTime": "2022-06-27T13:00:00.003Z",
            "weekday": "FRI"
          },
           {
            "startTime": "2022-06-27T08:00:00.003Z",
            "endTime": "2022-06-27T16:00:00.003Z",
            "weekday": "WED"
          },
          "frequency": "NONE",
          "startDate": "2022-05-25T08:00:00.003Z",
          "endDate": "2022-07-22T08:00:00.003Z",
          "propertyHandle": "prop2"
        } */

        {
          startTime: '2022-08-19T08:00:00.003Z',
          endTime: '2022-08-19T16:00:00.003Z',
          propertyId: prop2.id,
        },
      ],
    }),
  ]);
}
