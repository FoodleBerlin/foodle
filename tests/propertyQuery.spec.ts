import { clean } from '../utils/clean';
import { seed } from '../utils/seed';

beforeAll(async () => {
  await clean();
  await seed();
});

describe(' Property', () => {
  it('can query a single property by the handle', async () => {
    /*   const query = `
    query Query($handle: String) {
      findProperty(handle: $handle) {
        ClientErrorInvalidHandle {
          message
        }
        Property {
          city
        }
      }
    }
    `;
    const res = await server.executeOperation({
      query,
      variables: { handle: '1' }, //TODO get this from globals
    });
    expect(res).toMatchSnapshot(); */
  });

  /*  it('can query a list of multiple properties', async () => {
    const query = `
    query Query {
      findAllProperties {
        Properties {
          city
        }
        UnknownError {
          message
        }
      }
    }
    `;
    const res = await server.executeOperation({
      query,
    });
    expect(res).toMatchSnapshot();
  }); */
});
