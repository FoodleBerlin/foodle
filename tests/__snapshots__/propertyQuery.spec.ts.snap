// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Property can query a list of multiple properties 1`] = `
Object {
  "data": Object {
    "findAllProperties": Object {
      "Properties": Array [
        Object {
          "city": "Germany2",
        },
      ],
      "UnknownError": null,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[` Property can query a single property by the handle 1`] = `
Object {
  "data": Object {
    "findProperty": Object {
      "ClientErrorInvalidHandle": null,
      "Property": Object {
        "city": "Berlin",
      },
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
